mavros_extras/src/plugins/trajectory.cpp

```diff
@@ -73,175 +81,160 @@ class TrajectoryPlugin : public plugin::PluginBase {
 		trajectory.time_usec = req->header.stamp.toNSec() / 1000;	//!< [milisecs]
 		trajectory.type = req->type;	//!< trajectory type (waypoints, bezier)
 
-		/* Transformation from ENU to NED */
+		auto fill_points_position = [] (std::array<float, LEN> &point_output, const float x, const float y, const float z) {
+			// [[[cog:
+			//     cog.outl("auto position_ned = ftf::transform_frame_enu_ned(Eigen::Vector3d(x, y, z));")
+			//     for index, axis in zip ("012", "xyz"):
+			//         cog.outl("point_output[{index}] = position_ned.{axis}();".format(**locals()))
```

Comment:
I think you can define function for that `outl()`. Else it is not necessary to include first line.
Also better to use ros geometry_msgs / Vector / Point as input. And anyway it should be `double`.

Reply:
Ok, I'll revert back to have each point in `Trajectory` to be a `mavros_msgs::PositionTarget` and have the time_horizon for bezier in the Trajectory. I changed everything to a general array because the 4th element of the array is either `vx` or `time_horizon` depending on the type. I totally see your point of using the mavros geometry_msgs etc and in fact that was my initial choice.